x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/ApiKey.java

```diff
@@ -265,6 +274,39 @@ public XContentBuilder innerToXContent(XContentBuilder builder, Params params) t
         return builder;
     }
 
+    private void buildXContentForCrossClusterApiKeyAccess(XContentBuilder builder, RoleDescriptor roleDescriptor) throws IOException {
```

Comment:
This is very optional: I don't think we have an explicit "roundtrip" test (aside from the update API key REST test), i.e., something like (note, this just shows the idea, it's not fully functional):

```java
public void testRoundTrip() {
	final RoleDescriptor roleDescriptor = randomCrossClusterAccessRoleDescriptor();
	XContentBuilder b = jsonBuilder();
	...
	ApiKey.buildXContentForCrossClusterApiKeyAccess(b, roleDescriptor);
	...
	final CrossClusterApiKeyRoleDescriptorBuilder access = parseForAccess(Strings.toString(b));
	assertThat(access.build(), equalTo(roleDescriptor));
}
```

I think it _could_ be worth adding but also fine with skipping it since we have good test coverage already.

Reply:
We do have such a roundtrip assertion in [ApiKeyTests](https://github.com/elastic/elasticsearch/pull/97605/files#diff-8d6947077e1c4e4ea5021871195ab1cbc7073b0fecf962ada8a94781eb1f469aR117-R120).